### PR TITLE
API Gateway deployment is configured for caching and cache is not encrypted.

### PR DESCRIPTION
**Pull request by pethers**
https://github.com/stelligent/cfn_nag/issues/503

### CODE REVIEW

**Comment by pethers** [comment]
Should consider methods overrides in https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-stage-methodsetting.html#cfn-apigateway-stage-methodsetting-cachedataencrypted

File: spec/custom_rules/ApiGatewayCacheEncryptedRule_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+require 'cfn-model'
+require 'cfn-nag/custom_rules/ApiGatewayCacheEncryptedRule'
+
+describe ApiGatewayCacheEncryptedRule do
+  context 'Api Gateway has cache encryption enabled' do
+    it 'returns offending logical resource id' do
```
Comment (arothian) [inline spec/custom_rules/ApiGatewayCacheEncryptedRule_spec.rb:7]: Minor recommended change to match test case functionality
```suggestion
    it 'returns no violating resources' do
```

File: spec/custom_rules/ApiGatewayCacheEncryptedRule_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+      actual_logical_resource_ids = ApiGatewayCacheEncryptedRule.new.audit_impl cfn_model
+      expected_logical_resource_ids = %w[]
+
+      expect(actual_logical_resource_ids).to eq expected_logical_resource_ids
+    end
+  end
+
+  context 'Api Gateway has no cache configured' do
+    it 'returns offending logical resource id' do
```
Comment (arothian) [inline spec/custom_rules/ApiGatewayCacheEncryptedRule_spec.rb:18]:
```suggestion
    it 'returns no violating resources' do
```

File: spec/custom_rules/ApiGatewayCacheEncryptedRule_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+      actual_logical_resource_ids = ApiGatewayCacheEncryptedRule.new.audit_impl cfn_model
+      expected_logical_resource_ids = %w[]
+
+      expect(actual_logical_resource_ids).to eq expected_logical_resource_ids
+    end
+  end
+
+  context 'Api Gateway with no cache encryption enabled' do
+    it 'returns empty' do
```
Comment (arothian) [inline spec/custom_rules/ApiGatewayCacheEncryptedRule_spec.rb:29]:
```suggestion
    it 'returns violating resource ids' do
```

**Comment by pethers** [comment]
@arothian thanks for providing the fixes, cut & paste errors I should have noticed.

**Review by arothian — APPROVED** [review]

**Comment by IlyasMohammad** [comment]
@pethers @arothian
I have ruby version "ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-darwin18]" 
I am trying to running cfn_nag_scan command from my Mac terminal.
I am getting the below error.  I tried 'gem pristine psych --version 3.2.0', but it's the same issue.
Can you please help?

`Ignoring psych-3.2.0 because its extensions are not built. Try: gem pristine psych --version 3.2.0
<internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- webrick/httputils (LoadError)
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/aws-sdk-s3-1.60.2/lib/aws-sdk-s3/legacy_signer.rb:5:in `<top (required)>'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/aws-sdk-s3-1.60.2/lib/aws-sdk-s3/customizations.rb:7:in `<top (required)>'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/aws-sdk-s3-1.60.2/lib/aws-sdk-s3.rb:36:in `require_relative'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/aws-sdk-s3-1.60.2/lib/aws-sdk-s3.rb:36:in `<top (required)>'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/lib/cfn-nag/rule_repos/s3_based_rule_repo.rb:3:in `<top (required)>'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/lib/cfn-nag/custom_rule_loader.rb:8:in `require_relative'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/lib/cfn-nag/custom_rule_loader.rb:8:in `<top (required)>'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/lib/cfn-nag/cfn_nag.rb:3:in `require_relative'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/lib/cfn-nag/cfn_nag.rb:3:in `<top (required)>'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/lib/cfn-nag.rb:3:in `<top (required)>'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/usr/local/Cellar/ruby/3.0.0_1/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7/bin/cfn_nag_scan:4:in `<top (required)>'
	from /usr/local/bin/cfn_nag_scan:6:in `load'
	from /usr/local/bin/cfn_nag_scan:6:in `<main>'`

**Comment by pethers** [comment]
Think it is unrelated to this PR, but you are running an old version " /usr/local/Cellar/gem-cfn-nag/0.6.7/gems/cfn-nag-0.6.7" 

Use latest version 0.7.2 that include ruby3 fixes in https://github.com/stelligent/cfn_nag/releases/tag/v0.7.1 and https://github.com/stelligent/cfn_nag/releases/tag/v0.7.0

**Pull request merged**